### PR TITLE
Remove the non-core parts of the API

### PR DESCRIPTION
**Pull request by IvanPombo**
This PR removes all the non-core parts of the API.

All of the scenarios are running in [here](https://github.com/inductiva/inductiva-scenarios), with these change they won't be working in this branch. However, we will implement functionalities to simplify the user scenario creation.
In case you want to keep developing your scenarios, we advise you to bring that code to an individual repository and start building from there. As long as you install from the main inductiva, everything will be fine. However, it may break in the near future.

Removed:
- All of the scenarios;
- All of the template files;
- All visualization tools;
- All dependency management and all of the extra functions to help run the scenarios.

Added:
- All of template mechanics are introduced with a mixin `FileManager`;

This work is associated with the inductiva/tasks#3.

This serves to start working on the new functionalities with a clean slate.

### CODE REVIEW

**Comment by IvanPombo** [comment]
@sergiomsantos shall we merge this soon, to start working on the main package?

File: inductiva/mixins/file_manager.py
Comment (IvanPombo) [reply inductiva/mixins/file_manager.py:1]: Yes, I did mean it because I want to check that the FileManager and the removal was worked as expected to merge into development.

Changed the description to accomodate this.

**Review by sergiomsantos — APPROVED** [review]

File: inductiva/tasks/task.py
Comment (AugustoPeres) [inline inductiva/tasks/task.py:229]: Why were these methods removed? Where were they moved to?

File: inductiva/mixins/file_manager.py
Comment (IvanPombo) [reply inductiva/mixins/file_manager.py:210]: This was a mistake.

File: inductiva/tasks/task.py
Comment (IvanPombo) [reply inductiva/tasks/task.py:229]: They were removed because they were attached to the scenarios existence. Their sole purpose was to fetch the output class of the scenarios.

**Review by sergiomsantos — CHANGES_REQUESTED** [review]
The README needs to be updated accordingly

File: inductiva/mixins/file_manager.py
Comment (IvanPombo) [reply inductiva/mixins/file_manager.py:1]: Exactly, it is just to be a plug-play within scenarios without being the parent of the created class.

File: inductiva/mixins/file_manager.py
Comment (IvanPombo) [reply inductiva/mixins/file_manager.py:25]: Technically, it executes file management operations and can be used in general to manage files, however it is true that it encapsulates both just adding/copying files and rendering at the same time. We started with separate entities, both the code and structure of them was intrinsically similar so we decided to converge into one.

With this in mind, point 1) and 2) are just possible applications of this manager and we could also use it for other aspects, like moving files after the simulation finish into a single folder, or even manage the datasets from the simulation outputs. So I think the name is still appropriate for our context.

File: inductiva/mixins/file_manager.py
Comment (sergiomsantos) [reply inductiva/mixins/file_manager.py]: Good catch here. Probably a leftover from when we had the `render_file`/`render_dir` methods 🤔 !?

File: inductiva/mixins/file_manager.py
Comment (sergiomsantos) [reply inductiva/mixins/file_manager.py:25]: > Technically, it executes file management operations and can be used in general to manage files, however it is true that it encapsulates both just adding/copying files and rendering at the same time. We started with separate entities, both the code and structure of them was intrinsically similar so we decided to converge into one.

I support this reasoning here. We had the `add_file|dir` and `render_file|dir` primitives and the end result was that everything was ver similar amongst the render/add groups.

As to having this mixin to manage outputs, I'm not sure it would be a valid use case for it.

File: inductiva/mixins/file_manager.py
Comment (sergiomsantos) [inline inductiva/mixins/file_manager.py:93]: We can give a bit more information. Also, a few examples would help.

```suggestion
        """Add a file to the root directory and render it in case it is a template. 
        
        This method copies the contents of the `source_file` to a `target_file`
        inside the root directory. If the source file is a template file, it is rendered
        using the given `render_args` before copying to the target destination.
        If `target_file` is None, the copied file is saved inside the manager's root
        directory with the same name as the source file; if it is a template file, the
        template suffix is removed. 
        A file is considered to be a template file if it ends with the `.jinja` suffix.
        The rendering arguments are ignored if the source file is not a template
        file.

        Args:
            source_file: Path to the source file.
            target_file: Path to the target file.
            render_args: Arguments for rendering the template file.
        
        Examples:
        
        <Add 2 or 3 examples here>
        """
```

File: inductiva/mixins/file_manager.py
Comment (sergiomsantos) [inline inductiva/mixins/file_manager.py:130]: same here

```suggestion
        """Add a directory to the root directory and render all template files. 
        
        This method copies the contents of the `source_dir` to a `target_dir`
        inside the root directory, keeping the structure as the source directory.
        All template files in the source directory are rendered
        using the given `render_args` before copying to the target destination.
        
        If `target_dir` is None, the source directory is copied to the root
        directory. All template files will have the template suffix stripped from their
        name upon copying.
        A file is considered to be a template file if it ends with the `.jinja` suffix.
        The rendering arguments are ignored no template file is available in the
        source directory.

        Args:
            source_dir: Path to the source directory.
            target_dir: Path to the target directory.
            render_args: Arguments for rendering the template files.
        
        Examples:
        
        <Add 2 or 3 examples here>
        """
```

**Review by sergiomsantos — COMMENTED** [review]
let's add a few examples to each add_file/dir method and enhance the description before merging to dev.

File: inductiva/mixins/file_manager.py
Comment (sergiomsantos) [inline inductiva/mixins/file_manager.py:108]:
```suggestion
        a) Add a file to the root directory "example_a"
            (input_file.txt -> example_a/input_file.txt):
            >>> file_manager = FileManager()
            >>> file_manager.set_root_dir("example_a")
            >>> file_manager.add_file("input_file.txt")
        
        b) Add a file to the root directory "example_b" with a different name
            (input_file.txt -> example_b/renamed_file.txt):
            >>> file_manager = FileManager()
            >>> file_manager.set_root_dir("example_b")
            >>> file_manager.add_file("input_file.txt", "renamed_file.txt")
        
        c) Render a template file to the root directory "example_c" with a
            template argument named "value" (input_file.txt.jinja ->
            example_c/input_file.txt):
            >>> file_manager = FileManager()
            >>> file_manager.set_root_dir("example_c")
            >>> file_manager.add_file("input_file.txt.jinja", value=2)
```

File: inductiva/mixins/file_manager.py
Comment (sergiomsantos) [inline inductiva/mixins/file_manager.py:171]:
```suggestion
        a) Add a directory to the root directory "example_a"
            (input_dir/ -> example_a/input_dir/):
            >>> file_manager = FileManager()
            >>> file_manager.set_root_dir("example_a")
            >>> file_manager.add_dir("input_dir")

        b) Add directory to the root directory "example_b" with different name
            (input_dir/ -> example_b/renamed_dir/):
            >>> file_manager = FileManager()
            >>> file_manager.set_root_dir("example_b")
            >>> file_manager.add_dir("input_dir", "renamed_dir")
        
        c) Render template directory to the root directory with the template
            arguments named "value" and "name" (input_dir/ -> example_c/input_dir/):
            >>> file_manager = FileManager()
            >>> file_manager.set_root_dir("example_c")
            >>> file_manager.add_dir("input_dir", value=2, name="example")
```

**Review by sergiomsantos — COMMENTED** [review]
just a minor correction to the examples. I'll approve it after

**Review by sergiomsantos — APPROVED** [review]
.

**Pull request merged**
